### PR TITLE
Extend Interop

### PR DESCRIPTION
**Pull request by nodh**
Further improve on EUDI interoperability, regarding SIOPv2 and OID4VCI

### CODE REVIEW

File: .gitconfig
Comment (JesusMcCloud) [inline .gitconfig:1]: :+1:

File: vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/OidcUserInfoExtended.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import at.asitplus.KmmResult
+import at.asitplus.KmmResult.Companion.wrap
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.decodeFromJsonElement
+import kotlinx.serialization.json.encodeToJsonElement
+
+/**
+ * Holds a deserialized [OidcUserInfo] as well as a [JsonElement] with other properties,
```
Comment (JesusMcCloud) [inline vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/OidcUserInfoExtended.kt:11]: The code says `JsonObject`

File: vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/SupportedCredentialFormat.kt
Comment (JesusMcCloud) [inline vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/SupportedCredentialFormat.kt:1]: I get the encasulation part to store it as a single property and expose ISO and sdJwt separately. but why are these mutable?

File: vclib/src/commonMain/kotlin/at/asitplus/wallet/lib/data/DurationSecondsIntSerializer.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import kotlinx.serialization.encoding.Encoder
+import kotlin.time.Duration
+
+object DurationSecondsIntSerializer : KSerializer<Duration> {
+
+    override val descriptor: SerialDescriptor =
+        PrimitiveSerialDescriptor("DurationSecondsIntSerializer", PrimitiveKind.LONG)
+
+    override fun deserialize(decoder: Decoder): Duration {
+        return Duration.parse("${decoder.decodeInt()}s")
```
Comment (JesusMcCloud) [inline vclib/src/commonMain/kotlin/at/asitplus/wallet/lib/data/DurationSecondsIntSerializer.kt:17]: I know it is functionally equivalent, but wouldn't `decoder.decodeInt().seconds` be more concise, straight-forward and safe a virtually useless string instantiation?

**Review by JesusMcCloud — CHANGES_REQUESTED** [review]
nice work! minor polishing requested

File: vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/SupportedCredentialFormat.kt
Comment (nodh) [reply vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/SupportedCredentialFormat.kt:1]: To be able to create the claims in the correct format when constructing the object. I'm also not happy with the setter, but with it being a data class and so on ...

Comment (JesusMcCloud) [reply vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/SupportedCredentialFormat.kt:1]: i still don't get it. why not add a secondary constructor taking iso claims and sd-jwt claims as parameters and just keep the custom getters for those two properties. Then again, I'm probably lacking context

File: vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/SupportedCredentialFormat.kt [hunk 3]
```diff
@@ -107,19 +107,90 @@ data class SupportedCredentialFormat(
...
-            jsonSerializer.decodeFromJsonElement<Map<String, RequestedCredentialClaimSpecification>>(it)
-        }
-        set(value) {
-            claims = jsonSerializer.encodeToJsonElement(value)
+            runCatching {
+                jsonSerializer.decodeFromJsonElement<Map<String, RequestedCredentialClaimSpecification>>(it)
+            }.getOrNull()
         }
+
+    companion object {
```
Comment (JesusMcCloud) [inline vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/SupportedCredentialFormat.kt:124]: if all these are mutually exclusive (which I understand they are), then this is the way to go!

**Review by JesusMcCloud — APPROVED** [review]
ship it!

File: vclib-openid/src/commonTest/kotlin/at/asitplus/wallet/lib/oidvci/OidvciProcessTest.kt [hunk 4]
```diff
@@ -94,7 +96,9 @@ class OidvciProcessTest : FunSpec({
         val jws = JwsSigned.parse(serializedCredential.substringBeforeLast("~"))
         jws.shouldNotBeNull()
         val sdJwt = VerifiableCredentialSdJwt.deserialize(jws.payload.decodeToString())
-        sdJwt.shouldNotBeNull().also { println(it) }
+            .getOrThrow().shouldNotBeNull()
+
+        println(sdJwt)
```
Comment (acrusage) [inline vclib-openid/src/commonTest/kotlin/at/asitplus/wallet/lib/oidvci/OidvciProcessTest.kt:101]: Is `println` better than using `Napier.d` here?

File: vclib-openid/src/commonTest/kotlin/at/asitplus/wallet/lib/oidvci/OidvciProcessTest.kt [hunk 5]
```diff
@@ -71,7 +71,9 @@ class OidvciProcessTest : FunSpec({
...
```
Comment (acrusage) [inline vclib-openid/src/commonTest/kotlin/at/asitplus/wallet/lib/oidvci/OidvciProcessTest.kt:76]: Is println better than using Napier.d here?

**Review by acrusage — APPROVED** [review]
I don't have much much input in terms of semantics, so here you go!

**Review by acrusage — CHANGES_REQUESTED** [review]
Better not merge into main from a feature branch

File: vclib-openid/src/commonTest/kotlin/at/asitplus/wallet/lib/oidvci/OidvciProcessTest.kt (hunk 4 above)
Comment (nodh) [reply vclib-openid/src/commonTest/kotlin/at/asitplus/wallet/lib/oidvci/OidvciProcessTest.kt:101]: In tests it doesn't matter really, I think

**Review by acrusage-iaik — APPROVED** [review]

**Pull request merged**